mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraUidProvider.java

```diff
@@ -101,66 +91,54 @@ public MessageUid nextUid(MailboxSession mailboxSession, Mailbox mailbox) throws
     public MessageUid nextUid(MailboxSession session, MailboxId mailboxId) throws MailboxException {
         CassandraId cassandraId = (CassandraId) mailboxId;
         return nextUid(cassandraId)
-        .join()
+        .blockOptional()
         .orElseThrow(() -> new MailboxException("Error during Uid update"));
     }
 
-    public CompletableFuture<Optional<MessageUid>> nextUid(CassandraId cassandraId) {
-        return findHighestUid(cassandraId)
-            .thenCompose(optional -> {
-                if (optional.isPresent()) {
-                    return tryUpdateUid(cassandraId, optional);
-                }
-                return tryInsert(cassandraId);
-            })
-            .thenCompose(optional -> {
-                if (optional.isPresent()) {
-                    return CompletableFuture.completedFuture(optional);
-                }
-                return runner.executeAsyncAndRetrieveObject(
-                    () -> findHighestUid(cassandraId)
-                        .thenCompose(readUid -> tryUpdateUid(cassandraId, readUid)));
-            });
+    public Mono<MessageUid> nextUid(CassandraId cassandraId) {
```

Comment:
Why can't I see `handlesRetry method` in this class?

Reply:
Does this means we do not retry UID creation ???

Damn that explain some cassandra test random failures (MailboxManagerStressTest)